### PR TITLE
feat: replace native pattern select with dropdown

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
Replaces the pattern select with the more accessible Dropdown component

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [n/a] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

| Before | After |
|--------|-------|
|image|image|

### CODE REVIEW

File: percy/percy.test.js [hunk 1]
```diff
@@ -390,36 +397,32 @@ test('OTP-RR', async () => {
...
+  await page.keyboard.press('Enter')
+
   await page.waitForTimeout(1000)
 
   await percySnapshotWithWait(page, 'Pattern Viewer Showing Route 1')
 
   // Stop viewer from pattern viewer
   try {
-    await page.$x("//a[contains(., 'West')]")
+    await page.$x("//button[contains(@name, 'West')]")
```
Comment (miles-grant-ibigroup) [inline percy/percy.test.js:421]: Wow well done on figuring this one out

File: percy/percy.test.js [hunk 2]
```diff
@@ -164,20 +164,27 @@ if (OTP_RR_PERCY_MOBILE) {
...
+
+    await page.click('#headsign-selector-label')
+
+    await page.waitForTimeout(500)
+
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Enter')
```
Comment (miles-grant-ibigroup) [inline percy/percy.test.js:179]: Oy vey... are we sure we don't have a better solution here? Or can we at least add a comment about what's going on

File: lib/components/util/dropdown.tsx [hunk 3]
```diff
@@ -169,7 +173,7 @@ export const Dropdown = ({
         tabIndex={0}
         title={label}
       >
-        {name}
+        <NameWrapper>{name}</NameWrapper>
```
Comment (miles-grant-ibigroup) [inline lib/components/util/dropdown.tsx:176]: Are we expecting this to be styled in the future? Why not just make it a `span`?

**Review by miles-grant-ibigroup — APPROVED** [review]
Working great thanks so much!

**Comment by binh-dam-ibigroup** [comment]
I see some flashing when switching patterns, can that be addressed?

**Comment by binh-dam-ibigroup** [comment]
> I see some flashing when switching patterns, can that be addressed?

Basically, the route viewer flashes briefly during the transition from one pattern to another.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Can you address the flashing, please?

File: lib/components/viewers/route-details.js [hunk 4]
```diff
@@ -78,6 +80,14 @@ class RouteDetails extends Component {
 
     const routeColor = getRouteColorBasedOnSettings(operator, route)
 
+    const PatternSelectButton = styled(UnstyledButton)`
+      span {
+        white-space: nowrap;
+        overflow: hidden;
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/route-details.js:86]: Sort CSS

File: lib/components/viewers/styled.js [hunk 5]
```diff
@@ -33,13 +33,29 @@ export const PatternContainer = styled.div`
...
+  }
+
+  // Styling for SortResultsDropdown
+
+  & > span {
+    width: 85%;
+
+    button {
+      display: flex;
+      align-items: center;
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/styled.js:46]: Sort CSS

**Comment by amy-corson-ibigroup** [comment]
Flashing fixed in [dd8e99e](https://github.com/opentripplanner/otp-react-redux/pull/937/commits/dd8e99e3465a6586184c51970e58f643e7317b90), CSS in [a505a18](https://github.com/opentripplanner/otp-react-redux/pull/937/commits/a505a18c2a936d4d5a2c736e08bff2ae960342d5)!

File: lib/components/viewers/styled.js [hunk 6]
```diff
@@ -33,13 +33,29 @@ export const PatternContainer = styled.div`
...
+    button {
+      align-items: center;
+      display: flex;
+      justify-content: space-between;
+      max-width: 300px;
+      width: 95%;
+
+      span {
+        text-overflow: ellipsis;
+        overflow: hidden;
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/styled.js:53]: Sort CSS.

File: lib/reducers/create-otp-reducer.js [hunk 7]
```diff
@@ -736,7 +736,6 @@ function createOtpReducer(config) {
           // If setting to a route (not null), also set main panel.
           return update(state, {
             ui: {
-              mainPanelContent: { $set: MainPanelContent.ROUTE_VIEWER },
```
Comment (binh-dam-ibigroup) [inline lib/reducers/create-otp-reducer.js:739]: I am suspecting that this change is interfering with navigating with the back button. To see the effect:
1. Open the route viewer, select a route, and go to pattern view.
2. Click the browser Back button.
3. Observe that the route viewer returns to a state where no route is selected. The route selected in step 1 should remain selected.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Sorry @amy-corson-ibigroup, I noticed that with this PR, the ability to navigate back is affected (see comment).

File: lib/reducers/create-otp-reducer.js (hunk 7 above)
Comment (amy-corson-ibigroup) [reply lib/reducers/create-otp-reducer.js:739]: Great catch thank you! Fixed in [0f8aea1](https://github.com/opentripplanner/otp-react-redux/pull/937/commits/0f8aea157043848a60621d927701606d32f012ca)

File: lib/reducers/create-otp-reducer.js [hunk 8]
```diff
@@ -733,9 +733,14 @@ function createOtpReducer(config) {
 
       case 'SET_VIEWED_ROUTE':
         if (action.payload) {
-          // If setting to a route (not null), also set main panel.
+          // If setting to a route, assign the MainPanel according to whether there is a pattern.
+          const mainPanelValue =
+            action.payload && action.payload.patternId
```
Comment (binh-dam-ibigroup) [inline lib/reducers/create-otp-reducer.js:738]: Remove the `action.payload` condition, it is already defined per the if block.

File: lib/reducers/create-otp-reducer.js (hunk 7 above)
Comment (binh-dam-ibigroup) [reply lib/reducers/create-otp-reducer.js:739]: Great, much better now!

**Review by binh-dam-ibigroup — APPROVED** [review]
Thanks for the changes!

**Pull request merged**